### DISCUSSION TITLE
Wikipedia entries - is this available Offline

### DISCUSSION THREAD

**Discussion by pwb999** [organicmaps/organicmaps#2513] [Q&A] [answered]
Its great that Wikipedia information is assigned to some place names. I live in a hilly area and it seems that each peak has a Wiki entry. this is very useful whilst walking in the area.
However do I need a wifi connection to access the Wiki entries or is this information embeddened in the Country Downloads
I need to be sure that no internet connection is necessary to view this excellent source.

Thanks

**Comment by dbf256** [comment, marked as answer]
Yes, it is included into map files. To check you can disable internet access and load some new article.

**Reply by pwb999** [reply]
Excellent. This is really useful for hiking and great thats it is completely offline.
Many thanks

**Comment by GhostInTheKayak** [comment]
BTW -- how do you select which Wikipedia articles to include?
I see that many of my local locations do have Wikipedia articles, but the matching WP article is not available in OM.
If the WP article does have co-ordinates, how far adrift can they be to NOT be matched to the POI?

**Reply by biodranik** [reply]
Only OSM features with [wikipedia](https://wiki.openstreetmap.org/wiki/Key:wikipedia) and wikidata have Wiki articles.

**Reply by GhostInTheKayak** [reply]
Thanks.  So my mission should be to add WP links into OSM!

**Reply by GhostInTheKayak** [reply]
Sounded clear in principle, but it does not seem to be the case in practice.
For example:
-- see the OSM entry for _Pitlochry railway station_
OSM has Wikidata = Q2215314 and a WP link https://en.wikipedia.org/wiki/Pitlochry_railway_station
OM provides a link to en.m.wikipedia.org/Pitlochry railway station
OM **does** show a text summary of the WP article
-- see the OSM entry for _Pitlochry_
OSM has Wikidata = Q1011665 and a WP link https://en.wikipedia.org/wiki/Pitlochry
OM provides a link to en.m.wikipedia.org/Pitlochry
OM **does not** show a text summary of the WP article

I understand the change of link to provide a view of WP for mobiles (even though it is very annoying on an iPad).

However, why does the station show a WP summary text but the town does not?
Neither OSM entry has been edited in the last 4+ months.
BTW: The nearby town of Kinnaird **does** show a text summary of the WP article.

**Reply by biodranik** [reply]
@newsch can you please check it?

**Reply by newsch** [reply]
I can confirm that the OSM tags are recognized and the wikipedia article is processed.

The description is present when selecting Pitlochry on the map but **not** when searching for it.

This seems to be an issue with the "World" map getting priority over downloaded sections, which doesn't include descriptions AFAIK.

I can reproduce the behavior by selecting while zoomed out or searching for _any entry that is present in the world map **and** the local map_, and has a wikipedia description.

If I select Pitlochry on the map (while zoomed in) after downloading the area I see the description:
![Screenshot_20240207-144434](https://github.com/organicmaps/organicmaps/assets/10699964/35a9be3a-6f9b-42a4-8368-31a24f758ea3)

But if I search for Pitlochry and select the result, it zooms out and I don't see the description:
![Screenshot_20240207-151437](https://github.com/organicmaps/organicmaps/assets/10699964/27b6766b-1b25-4dbf-a277-9fb5611f8bda)

**Comment by GhostInTheKayak** [comment]
:-(
My experience of this is different - the town of Pitlochry has a WP link but no description - regardless of how I find Pitlochry (search or hunting on the map and zoom in).
I am using an iPad. Version 2024.01.09-5. Data 5 January 2024.
I will update the maps and check if anything changes.

**Reply by newsch** [reply]
I'm using the latest Android F-Droid version which is still 2023.12.20-4. I'll try the latest version on an emulator.

**Reply by newsch** [reply]
On `2024.02.06-11-Web` I get the same results as above: with the region downloaded, the description is visible when I'm zoomed in and select Pitlochry, missing when I select it from search. Same with Aberfeldy to the southwest.

**Comment by GhostInTheKayak** [comment]
Sorry to take so long – I have been away in Scotland (including Pitlochry).
I don't really speak Objective-C – so I did not get far by hunting through the code.

If I understand correctly, the problem is the merging of information from the "world" map with the information from the "local" map.
(I assume that the world map has less information for a POI, that the local map would.)
Currently, one map can take priority over the other, so an empty WP description can have priority over a defined one.
This is the case for Pitlochry town – the world map has no WP description, but is being given priority over the local map which does have a WP description. The world maps wins and no description is displayed.

So the fix would be to instead, if the WP description is present in one, but not the other, the "merge info" code needs to accept the non-empty value – whichever map it came from. Not sure which would get priority if both maps have the description. Ideally, it would be the one with the newer date.

Does my summary make any sense?

**Reply by newsch** [reply]
Thanks for looking into it more, sorry if I gave the wrong impression about my certainty of the cause. The world map vs. local map priority is just a hunch, and I don't know if the search implementation is platform-specific. I think we should:
1. Verify that the unexpected behavior is still present.
    - I'm still getting the same behavior on Android, and it acts the same on the Linux desktop beta. I don't have any iOS devices, but I can try to use a friend's. 
2. Clarify the differences on Android/iOS.
    - You're saying you never see an offline Wikipedia description for Pitlochry, but you do with other nearby towns? What about the nearby spots Perth and "Scotland", do they have an offline description if you search or select? They exhibit the same behavior on my end.
3. Make one or multiple issues on the main repository with that info, and see what the developers think. I don't think they browse the discussions as frequently.
    - I can make this once we have all the info.

@biodranik does that sound reasonable?
